### PR TITLE
Support `SparseTensor` label in `LightGCN`

### PR DESCRIPTION
**Pull request by hzg0601**
fix the bug of models/lightgcn when edge_indx is an instance of SparseTensor in forward method

### CODE REVIEW

File: torch_geometric/nn/models/lightgcn.py [hunk 1]
```diff
@@ -110,7 +111,21 @@ def forward(self, edge_index: Adj,
                 :obj:`edge_index` will be used instead. (default: :obj:`None`)
         """
         if edge_label_index is None:
-            edge_label_index = edge_index
+            if isinstance(edge_index, SparseTensor):
+                edge_label_index = edge_index.to_torch_sparse_coo_tensor(
```
Comment (rusty1s) [inline torch_geometric/nn/models/lightgcn.py:115]:
```suggestion
                edge_label_index = torch.stack(edge_index.coo()[:2], dim=0)
```

File: torch_geometric/nn/models/lightgcn.py [hunk 2]
```diff
@@ -110,7 +111,21 @@ def forward(self, edge_index: Adj,
...
+                ).coalesce().indices()
+            elif isinstance(edge_index, torch.Tensor) and edge_index.is_sparse(
```
Comment (rusty1s) [inline torch_geometric/nn/models/lightgcn.py:117]: I don‘t think we need to support this case right now. Any `tensor.is_sparse` logic should be dropped here.

**Review by rusty1s — APPROVED** [review]
Thank you!
